traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/dns/ZoneManager.java

```diff
@@ -453,6 +454,17 @@ private static List<Record> fillZones(final Map<String, List<Record>> zoneMap, f
 		return records;
 	}
 
+	protected static String getTRLocalHostname(final TrafficRouter tr) throws UnknownHostException {
+		// if there is only one TR in the CRConfig, just use that TR's hostname
+		// instead of checking for the local server's hostname
+		final boolean singleTR = Iterators.size(tr.getCacheRegister().getTrafficRouters().fieldNames()) == 1;
```

Comment:
> would not be `null` or explicitly `undefined`

`undefined` is not a value in JSON, that's just JS. You'll never see anything "explicitly `undefined`" because that's not syntactically valid.